pkg/controller/storageprofile-controller.go

```diff
@@ -44,8 +54,24 @@ func (r *StorageProfileReconciler) Reconcile(_ context.Context, req reconcile.Re
 	if err := r.client.Get(context.TODO(), req.NamespacedName, storageClass); err != nil {
 		return reconcile.Result{}, err
 	}
+	if _, err := r.reconcileStorageProfile(storageClass); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Alert for incomplete profiles
+	numIncomplete := 0
+	storageProfileList := &cdiv1.StorageProfileList{}
+	if err := r.client.List(context.TODO(), storageProfileList); err != nil {
+		return reconcile.Result{}, err
+	}
+	for _, profile := range storageProfileList.Items {
```

Comment:
It seems a bit inefficient for each storage profile update to go over all the list and check incomplete. Can't we change so this alert can be accumulative and then only need to increase incomplete in case the storage profile that was created/updated was incomplete and decrease from it in case there was an update and now it is? WDYT?

Reply:
Regarding the list, if we want all StorageProfiles they should already be cached
because this controller is watching all of them, so the List call should be cheap

The challenge, if I go accumulative, can be
- Profile 1 is broken
- Metric value increased
- Profile 2 gets created and is completely correct
- Metric decreased, we stop alerting (But there's still a faulty profile)

Reply:
thats why I said only on update, on  create you dont need to decrease and in update you have the previous to compate to

Reply:
or mark a profile as incomplete either on the status or in an internal map and then check for it before decreasing

Reply:
An internal map sounds good! then I can conditionally increment and decrement
(The same problem in the acc. scenario can occur with updates as well, consider two updates to the same profile)

Reply:
Thinking about it more - that would mean that we're keeping a map of all storage classes potentially,
which might not be as cheap as the list call

Reply:
but the list call is not the problem the problem is you go over all the list and check for each profile, which is surely less cheap then a map :)

Reply:
Just remember don't want to keep state in memory in the controller, since the controller could disappear at any time, and have to be recreated. Also keep in mind reconcile loops should be idem potent, so if I run the same thing with the same inputs I should get the same results.

Reply:
right a map is not suitable. its just saving that info in the storage profile itself. so if that seems an over kill for a metric than I guess this is ok